keps/sig-node/4680-add-resource-health-to-pod-status/README.md

```diff
@@ -120,57 +121,66 @@ The proposal is to introduce an additional field:
 ```
 // AllocatedResourcesStatus represents the status of various resources
 // allocated for this Pod.
-AllocatedResourcesStatus ResourcesStatus 
+// +featureGate=ResourceHealthStatus
+// +optional
+// +patchMergeKey=name
+// +patchStrategy=merge
+// +listType=map
+// +listMapKey=name
+AllocatedResourcesStatus []ResourceStatus `json:"allocatedResourcesStatus,omitempty" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,14,rep,name=allocatedResourcesStatus"`
+```
 
-type ResourcesStatus map[ResourceName]ResourceStatus
+In alpha (kubernetes 1.31) the `ResourceStatus` is defined as:
 
+```
 type ResourceStatus struct {
-    // map of unique Resource ID to its health.
-    // At a minimum, ResourceID must uniquely identify the Resource
-    // allocated to the Pod on the Node for the lifetime of a Pod.
-    // See ResourceID type for it's definition.
-    Resources map[ResourceID] ResourceHealth
-    
-    // allow to extend this struct in future with the overall health fields or things like Device Plugin version
+	// Name of the resource. Must be unique within the pod and match one of the resources from the pod spec.
+	// +required
+	Name ResourceName `json:"name" protobuf:"bytes,1,opt,name=name"`
```

Comment:
Is "name" here, the extended resource name? Does it indicate which container it's associated with? Or is it the device name from the DRA plugin or CDI or whatever? What if two plugins have devices with the same name?

Reply:
oh, it is a good question. Do you have a suggestion for DRA? It was kind of obvious for Device Plugin

Reply:
For standard device plugin devices the name would match to this from the `resource_name` field in the pod-resources-api:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubelet/pkg/apis/podresources/v1/api.proto#L67-L71
```
message ContainerDevices {
    string resource_name = 1;
    repeated string device_ids = 2;
    TopologyInfo topology = 3;
}
```

The equivalent for DRA is:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubelet/pkg/apis/podresources/v1/api.proto#L67-L71
```
message DynamicResource {
    string claim_name = 2;
    string claim_namespace = 3;
    repeated ClaimResource claim_resources = 4;
}

message ClaimResource {
    repeated CDIDevice cdi_devices = 1 [(gogoproto.customname) = "CDIDevices"];
    string driver_name = 2;
    string pool_name = 3;
    string device_name = 4;
}

message CDIDevice {
    string name = 1;
}
```
So I agree it's non-obvious ...
Maybe a concatenation of `driver_name`, `pool_name`, and `device_name`?

Reply:
well, I assume this is already nested under a container status? Sorry, the whole status isn't here, I could go dig it up, but I imagine other readers will have the same question. I'm trying to see the whole struct. That's also why I asked about an example.

Since `ResourceHealth` has `ResourceID` which uniquely identifies the DRA device, and I am assuming this `AllocatedResourceStatus` is *container level*, I don't see the need for the name. However, that would leave devices with no container (e.g., network devices, which are pod-level) with nowhere to put their status.

So, the question is, is `AllocatedResourceStatus` pod-level or container-level? If it's pod-level, then I think we would need one entry for each request of each claim in DRA - then the name would be "claim name" + "request name".

Reply:
And if it's *container level*, we are missing somewhere to put resources that are pod level (AFAIK, that's only network devices)

Reply:
> So, the question is, is AllocatedResourceStatus pod-level or container-level?

Today it is container level. I will clarify and will add Pod level

Reply:
Ok, then name here is probably claim name + request name, since ResourceID has the full device path (and there is possibly more than one device per request)

Reply:
Updated to `dra:claim_namespace/claim_name`. What do you mean by the request? Happy to update

Reply:
`claim_namespace` is the same as the pod namespace so shouldn't be needed. Each claim consists of multiple, named requests, which is what I was referring to.

That said, I think we should just make this a constant, `claims`. The reason is that I don't want to impose the need to maintain this state (claim/request -> device ID) if they don't need to for any other reason. They may be doing it now, but if not, I wouldn't want to force them to. Using "dra" won't really mean anything to an ordinary user, as that term isn't really used in the API anywhere. In the PodSpec, these are the devices associated with `resources.claims`, so I think "claims" works. Or "resourceClaims"

Reply:
made it just "claims" for now. I'll try to get feedback from it